app/src/main/java/com/github/geohunt/app/data/repository/bounties/ActiveBountiesRepository.kt

```diff
@@ -0,0 +1,54 @@
+package com.github.geohunt.app.data.repository.bounties
+
+import com.github.geohunt.app.data.network.firebase.toList
+import com.github.geohunt.app.data.repository.AuthRepositoryInterface
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ktx.snapshots
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+
+class ActiveBountiesRepository(
+        database: FirebaseDatabase,
+        private val authRepository: AuthRepositoryInterface,
+        private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) : ActiveBountiesRepositoryInterface {
+    private val bountiesOfRef = database.getReference("bounties/ofUser")
+
+    override suspend fun joinBounty(bid: String) {
+        updateBountyState(bid,true)
+    }
+
+    override suspend fun leaveBounty(bid: String) {
+        updateBountyState(bid,false)
+    }
+
+    private suspend fun updateBountyState(bid: String, newState: Boolean) = withContext(ioDispatcher) {
+        authRepository.requireLoggedIn()
+        @Suppress("DEPRECATION") val uid = authRepository.getCurrentUser().id
+
+        val newVal = if (newState) true else null
+        bountiesOfRef
+                .child(uid)
+                .child(bid)
+                .setValue(newVal)
+                .await()
+    }
+
+    override fun getBounties(): Flow<List<String>> {
+        authRepository.requireLoggedIn()
+        @Suppress("DEPRECATION") val uid = authRepository.getCurrentUser().id
+
+        return bountiesOfRef
+                .child(uid)
+                .snapshots
+                .map {
+                    it.toList()
```

Comment:
The whole point of using the Map is to check whether the boolean is sets to true or false, so I believe it should be something like
```suggestion
                    it.children.mapNotNull{ it.key.takeIf{ it.getValue(Boolean::class.java) } }
```

Reply:
PS : I have done this code without Android Studio so try checking if it compiles 🙃

Reply:
This is already how .toList() is implemented (you can check FirebaseUtils.kt to convince yourself).